src/update_helper.rs

```diff
@@ -0,0 +1,199 @@
+//! # Functions to update timestamps.
+
+use crate::chat::{Chat, ChatId};
+use crate::contact::Contact;
+use crate::context::Context;
+use crate::param::{Param, Params};
+use anyhow::Result;
+
+impl Context {
+    /// Updates a contact's timestamp, if reasonable.
+    /// Returns true if the caller shall update the settings belonging to the scope.
+    /// (if we have a ContactId type at some point, the function should go there)
+    pub(crate) async fn update_contacts_timestamp(
+        &self,
+        contact_id: u32,
+        scope: Param,
+        new_timestamp: i64,
+    ) -> Result<bool> {
+        if let Ok(mut contact) = Contact::load_from_db(self, contact_id).await {
```

Comment:
Why return `Ok(false)` if `load_from_db` fails? `load_from_db` fail means there is something is wrong with the database or the `contact_id` is invalid, such error should be bubbled up with `?`.

Reply:
you're right.  may idea was that there may be invalid contact_ids for whatever reason and that should not fail the whole message, but i double-checked, that should really not happen in normal flow, i do a commit to change that.

Reply:
hm, in fact, `from_id` might be `0`, there is even a test for that, `test_no_from()` that fails if we just bubble up the error.

but it is probably still better to bubble up and to check for `from_id == 0` before calling`update_contacts_timestamp()`.